relay-sampling/src/lib.rs

```diff
@@ -809,79 +839,184 @@ impl Default for SamplingMode {
     }
 }
 
+/// Represents a list of rule ids which is used for outcomes.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct MatchedRuleIds(pub Vec<RuleId>);
+
+impl MatchedRuleIds {
+    /// Creates a MatchedRuleIds struct from a string formatted with the following format:
+    /// rule_id_1;rule_id_2;...
+    pub fn from_string(value: &str) -> Result<MatchedRuleIds, ParseIntError> {
+        let mut rule_ids = vec![];
+
+        for rule_id in value.split(';') {
```

Comment:
nit: Why a `;` and not a `,`? The outcome reason `Sampled:1,2,3` might be easier to read than `Sampled:1;2;3`